### PR TITLE
Deprecate Kount and UnionPay

### PR DESCRIPTION
**Pull request by scannillo**
### Summary of changes

- Deprecate Kount related methods
- Deprecate all UnionPay classes
- Silence deprecation warnings throughout project and demo using pragma marks
    - _Side note: sadly, there is no equivalent to this once we're in Swift_
- Add helpful messaging in `DEPRECATED_MSG_ATTRIBUTE` to provide merchant alternative solution

giphy

### Checklist

- [X] Added a changelog entry

### Authors
@scannillo

### CODE REVIEW

File: Demo/Application/Features/UnionPay/BraintreeDemoUnionPayViewController.m [hunk 1]
```diff
@@ -123,7 +123,10 @@ - (void)enroll:(__unused UIButton *)button {
                 
                 self.progressBlock(@"Tokenizing card");
                 
+#pragma clang diagnostic push
```
Comment (mattwylder) [inline Demo/Application/Features/UnionPay/BraintreeDemoUnionPayViewController.m:126]: Should we remove the UnionPay demo soon?

**Review by mattwylder — APPROVED** [review]

File: Demo/Application/Features/UnionPay/BraintreeDemoUnionPayViewController.m (hunk 1 above)
Comment (scannillo) [reply Demo/Application/Features/UnionPay/BraintreeDemoUnionPayViewController.m:126]: Yes! We should remove this demo when we remove UnionPay from the `next-major-version` branch 🥳

File: Sources/BraintreeCard/Public/BraintreeCard/BTCardClient.h [hunk 2]
```diff
@@ -68,7 +68,8 @@ typedef NS_ENUM(NSInteger, BTCardClientErrorType) {
 */
 - (void)tokenizeCard:(BTCardRequest *)request
              options:(nullable NSDictionary *)options
-          completion:(void (^)(BTCardNonce * _Nullable tokenizedCard, NSError * _Nullable error))completion;
+          completion:(void (^)(BTCardNonce * _Nullable tokenizedCard, NSError * _Nullable error))completion
+DEPRECATED_MSG_ATTRIBUTE("The UnionPay SMS integration is deprecated, as UnionPay can now be processed as a credit card through their partnership with Discover. Use `CardClient.tokenizeCard(card: completion:)`.");
```
Comment (jaxdesmarais) [inline Sources/BraintreeCard/Public/BraintreeCard/BTCardClient.h:72]: Small nit:
```suggestion
DEPRECATED_MSG_ATTRIBUTE("The UnionPay SMS integration is deprecated, as UnionPay can now be processed as a credit card through their partnership with Discover. Use `BTCardClient.tokenizeCard(card: completion:)`.");
```

File: Sources/BraintreeUnionPay/Public/BraintreeUnionPay/BTCardCapabilities.h [hunk 3]
```diff
@@ -3,6 +3,7 @@
 /**
  Contains information about a card's capabilities
  */
+DEPRECATED_MSG_ATTRIBUTE("The UnionPay SMS integration is deprecated, as UnionPay can now be processed as a credit card through their partnership with Discover. Use `CardClient.tokenizeCard(card: completion:)`.")
```
Comment (jaxdesmarais) [inline Sources/BraintreeUnionPay/Public/BraintreeUnionPay/BTCardCapabilities.h:6]: `CardClient` > `BTCardClient`

File: Sources/BraintreeUnionPay/Public/BraintreeUnionPay/BTCardClient+UnionPay.h [hunk 4]
```diff
@@ -11,6 +11,7 @@ NS_ASSUME_NONNULL_BEGIN
 /**
  BTCardClient category for UnionPay
  */
+DEPRECATED_MSG_ATTRIBUTE("The UnionPay SMS integration is deprecated, as UnionPay can now be processed as a credit card through their partnership with Discover. Use `CardClient.tokenizeCard(card: completion:)`.")
```
Comment (jaxdesmarais) [inline Sources/BraintreeUnionPay/Public/BraintreeUnionPay/BTCardClient+UnionPay.h:14]: `CardClient` > `BTCardClient`

File: Sources/BraintreeUnionPay/Public/BraintreeUnionPay/BTConfiguration+UnionPay.h [hunk 5]
```diff
@@ -7,6 +7,7 @@
 /**
  BTConfiguration category for UnionPay
  */
+DEPRECATED_MSG_ATTRIBUTE("The UnionPay SMS integration is deprecated, as UnionPay can now be processed as a credit card through their partnership with Discover. Use `CardClient.tokenizeCard(card: completion:)`.")
```
Comment (jaxdesmarais) [inline Sources/BraintreeUnionPay/Public/BraintreeUnionPay/BTConfiguration+UnionPay.h:10]: `CardClient` > `BTCardClient`

**Review by jaxdesmarais — APPROVED** [review]
👻

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,12 +1,18 @@
...
+  * UnionPay cards can now be processed as regular cards (through the `BraintreeCard` module) due to their partnership with Discover
+
 ## 5.17.0 (2022-12-05)
 * BraintreePayPalNativeCheckout (BETA)
   * Fix CocoaPods bug emitting "Cannot find interface declaration" error ([CocoaPods issue #11672](https://github.com/CocoaPods/CocoaPods/issues/11672))
   * Rename `riskCorrelationId` to `riskCorrelationID`
   * Rename `nativeRequest` to `request` internally in `tokenizePayPalAccount`
   * `tokenizePayPalAccount` now takes in a `request` of type `BTPayPalNativeRequest` instead of a `nativeRequest` of type `BTPayPalRequest`
 
+
```
Comment (scannillo) [inline CHANGELOG.md:15]:
```suggestion
```

**Pull request merged**